Add filter by a category, filter by confidence, display findings from a specific time (#826)
returntocorp/semgrep-docs: docs/semgrep-app/findings.md

| Filter      | Description  |
| ----------  | ------------ |
| **Projects**     | Filter by repositories connected to Semgrep App. |
| **Status**       | Filter for different findings triage states. Refer to the [following table](#triaging-findings) to understand triage states. |
| **Severities**    | Filter by the severity of a finding. Possible values: <ul><li>Low</li><li>Medium</li><li>High</li></ul> |
| **Actions**     | Filter by monitoring, commenting, or blocking rules in your Rule Board. |
| **Confidences** | Filter by indication of the rule to detect true positives. The higher the confidence the more true positives the rule may detect.  |
| **Categories**  |  Filter by various rule categories, such as security, or best practice rules. More than one category can be selected for filtering. |
| **Branches**     | Filter by findings in different Git branches. |
| **Rules**        | Filter by rules or rulesets that are included in your Rule Board. More than one rule can be selected for filtering. |
| **Rulesets**     | Filter by name of the ruleset where rules that matched the code belong. More than one rule or ruleset can be selected for filtering. |

> You can also filter for findings reported from specific time periods or group findings by rule. See [Displaying findings reported in a specific time](/semgrep-app/findings/#displaying-findings-reported-in-a-specific-time).

Author: ```suggestion
You can also filter for findings reported from specific time periods. See [Displaying findings reported in a specific time](/semgrep-app/findings/#displaying-findings-reported-in-a-specific-time).
```

| **Severities**    | Filter by the severity of a finding. Possible values: <ul><li>Low</li><li>Medium</li><li>High</li></ul> |
| **Actions**     | Filter by monitoring, commenting, or blocking rules in your Rule Board. |
| **Confidences** | Filter by indication of the rule to detect true positives. The higher the confidence the more true positives the rule may detect.  |
| **Categories**  |  Filter by various rule categories, such as security, or best practice rules. More than one category can be selected for filtering. |
| **Branches**     | Filter by findings in different Git branches. |
| **Rules**        | Filter by rules or rulesets that are included in your Rule Board. More than one rule can be selected for filtering. |
| **Rulesets**     | Filter by name of the ruleset where rules that matched the code belong. More than one rule or ruleset can be selected for filtering. |

You can also filter for findings reported from specific time periods or group findings by rule. See [Displaying findings reported in a specific time](/semgrep-app/findings/#displaying-findings-reported-in-a-specific-time).

To filter through findings:

> 1. Click the filter criteria.
> 1. The page then refreshes to reflect the additional criteria.
> 1. Optional: To futher refine your filter, select additional filters.

Author: ```suggestion
1. Click the filter criteria. The page then refreshes to reflect the additional criteria.
1. Optional: To further refine your filter, select additional filters.
```

    ![Findings information panel](/img/app-findings.png)<br />
    *Figure 2.* Findings information panel.

### Filtering findings

Filtering allows you to easily isolate groups of findings for ease in triaging and identifying related groups of issues. The following criteria are available for filtering:

| Filter      | Description  |
| ----------  | ------------ |
| **Projects**     | Filter by repositories connected to Semgrep App. |
| **Status**       | Filter for different findings triage states. Refer to the [following table](#triaging-findings) to understand triage states. |
| **Severities**    | Filter by the severity of a finding. Possible values: <ul><li>Low</li><li>Medium</li><li>High</li></ul> |
| **Actions**     | Filter by monitoring, commenting, or blocking rules in your Rule Board. |
| **Confidences** | Filter by indication of the rule to detect true positives. The higher the confidence the more true positives the rule may detect.  |
| **Categories**  |  Filter by various rule categories, such as security, or best practice rules. More than one category can be selected for filtering. |

> | **Branches**     | Filter by findings in different Git branches. More than one branch can be selected for filtering. |

Reviewer: We don't allow more than one yet. It's a bug we have filed

Author: ```suggestion
| **Branches**     | Filter by findings in different Git branches. |
```

> | **Rules**        | Filter by rules or rulesets that are included in your Rule Board. More than one rule can be selected for filtering. More than one rule can be selected for filtering. |

Reviewer: I think we repeated "More than one rule can be selected for filtering" twice here

Author: ```suggestion
| **Rules**        | Filter by rules or rulesets that are included in your Rule Board. More than one rule can be selected for filtering. |
```

| **Rulesets**     | Filter by name of the ruleset where rules that matched the code belong. More than one rule or ruleset can be selected for filtering. |

> You can also filter for findings reported from specific time periods or group findings by rule. For more information see 

Reviewer: see?

Author: ```suggestion
You can also filter for findings reported from specific time periods or group findings by rule. See [Displaying findings reported in a specific time](/semgrep-app/findings/#displaying-findings-reported-in-a-specific-time) and [Group by rule](/semgrep-app/findings/#grouping-by-rule) sections.
```

To filter through findings:

1. Click the filter criteria.
1. The page then refreshes to reflect the additional criteria.
1. Optional: To futher refine your filter, select additional filters.

#### Displaying findings reported in specific time

Display and filter for findings reported only in specific time. To display findings reported in a specific time, follow these steps:

1. In Semgrep App, go to **[Findings](https://semgrep.dev/orgs/-/findings)** page.
1. Click the <i class="fa-solid fa-calendar-days"></i> **All time** button.
1. Select preferred time period for which to display findings. The following options are available:
    - **All time** - This is the default option.
    - Last 1 year
    - Last 1 month
    - Last 7 days
    - Last 1 day

> ### Grouping by rule

Reviewer: This is still in beta. (Needs a feature flag) Not sure if we should include it yet?

Author: @annikalee ok...

Author: Solutions:
1. Remove this part of the documentation and keep it in another PR.
2. Mark this feature as a beta for specific customers who may request it.

Reviewer 2: @adamkvitek Channeling my inner Raghav here, I’d go for #2 but you may want to confirm with Chinmay.

Author: @neo413001 we were thinking with @s-santillan that solution 1. (see the conversation above in this thread) might be better for now. Should we mark this section about grouping by rule as beta or move it to another PR and wait until the feature is generally out for everyone? I think that if we mark it as beta we need to say that users need to contact us to get this beta feature.
